### PR TITLE
User can view a list of favorite streamers with real-time streaming details.

### PR DESCRIPTION
**Pull request by kenk2**
twitch-time

### CODE REVIEW

File: client/components/app.js [hunk 1]
```diff
@@ -26,22 +42,20 @@ export class App extends React.Component {
...
     return (
       <div>
-        <div className="search-bar">
-          <Search getChannel={this.getChannel} />
-        </div>
-        <div>
-          <SearchList list={this.state.list} addChannel={this.addChannel} />
-        </div>
+        <Search getChannel={this.getChannel} switchView={this.switchView}/>
+        {view}
```
Comment (victormejia) [inline client/components/app.js]: it's a little out of "React convention" to write it this way (with the `determineView`). Instead, do it inline here:

```jsx
{
  this.state.view === 'Search' 
    ? <SearchList list={component.state.list} addChannel={component.addChannel} />
    : <Favorites favorites />
}
```

File: client/components/favorites.js [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+    super(props)
+    this.state = {favorites: null}
+  }
+  componentDidMount() {
+    renderFavorites().then(res => {
+      return res.json()
+    })
+      .then(data => {
+        this.setState({favorites: data})
+      })
```
Comment (victormejia) [inline client/components/favorites.js]: fix indentation here (with the two .then's)

File: client/components/search.js [hunk 3]
```diff
@@ -70,25 +60,23 @@ export class Search extends React.Component {
             </IconButton>
             <Drawer
               docked={false}
-              width={200}
               open={this.state.open}
               onRequestChange={open => this.setState({open})}
             >
-              <MenuItem onClick={this.handleClose}>Favorites</MenuItem>
+              <MenuItem onClick={this.switchView}>Search</MenuItem>
```
Comment (victormejia) [inline client/components/search.js]: call the `switchView` function with the necessary data:

```js
onClick={() => this.switchView('Search')}
```

File: migrations/1502235319379_streamers.sql [hunk 4]
```diff
@@ -1,5 +1,5 @@
 -- up
-CREATE TABLE streamers (
+CREATE TABLE IF NOT EXISTS streamers (
```
Comment (victormejia) [inline migrations/1502235319379_streamers.sql]: you shouldn't need this with migrations.

**Review by victormejia — CHANGES_REQUESTED** [review]

**Comment by kenk2** [comment]
Approved by Tim.

**Pull request merged**